Source/Scene/Model.js

```diff
@@ -1453,6 +1453,103 @@ define([
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                                // Skip if the attribute is not used by the material, e.g., because the asset was exported
+                                // with an attribute that wasn't used and the asset wasn't optimized.
+                                if (defined(attributeVarName)) {
+                                    var a = accessors[primitiveAttributes[attributeSemantic]];
+                                    if (a.hasOwnProperty('extensions')) {
+                                        var extensions = a.extensions;
+                                        if (extensions.hasOwnProperty('WEB3D_quantized_attributes')) {
+                                            var quantizedAttributes = extensions.WEB3D_quantized_attributes;
+                                            var decodeMatrix = quantizedAttributes.decodeMatrix;
+                                            decodeMatrix = decodeMatrix.map(function(val) {
+                                                return val.toFixed(16);
+                                            });
+
+                                            var newMain = "decoded_" + attributeSemantic;
+                                            var matrixName = "decode_" + attributeSemantic + "_matrix";
+                                            var decodedAttributeVarName = attributeVarName.replace('a_', 'dec_');
+
+                                            // replace usages of the original attribute with the decoded version, but not the declaration
+                                            var first = true;
+                                            shader = shader.replace(new RegExp(attributeVarName, 'g'), function(match) {
+                                                if (first) {
+                                                    first = false;
+                                                    return match;
+                                                }
+                                                return decodedAttributeVarName;
+                                            });
+                                            // declare decoded attribute
+                                            shader = "vec3 " + decodedAttributeVarName + ";\n" + shader;
+
+                                            // splice decode function into the shader
+                                            var decode = "\n" +
+                                                         "mat4 " + matrixName + " = mat4(" + decodeMatrix + ");\n" +
+                                                         "void main(void) {\n" +
+                                                         "    " + decodedAttributeVarName + " = vec3(vec4(" + attributeVarName + ",1.0) * " + matrixName + ");\n" +
```

Comment:
I should probably make it so that we're pre-multiplying because its more consistent with how transformations are usually applied in shaders. 

Really the only reason it is like this was because I preferred to write the decodeMatrix in the gltf in row major form; that is another thing the spec needs to clarify.